crates/client/eth-client/src/config.rs

```diff
@@ -0,0 +1,141 @@
+//! Base Ethereum client configuration.
+//!
+//! Use it as is or reuse top-level fields in your config ("inherit")
+//! in order to share the same configuration file.
+//!
+//! struct MyServiceConfig {
+//!     pub provider: EthereumProviderConfig,
+//!     pub wallet: Option<EthereumWalletConfig>,
+//!     pub contracts: Option<StarknetContracts>,
+//! }
+//!
+//! Default provider and wallet configurations are set for use with Anvil
+//! - local Ethereum environment.
+
+use std::fs::File;
+use std::path::PathBuf;
+
+use ethers::types::Address;
+use serde::{Deserialize, Serialize};
+
+use crate::error::Error;
+
+/// Default Anvil local endpoint
+pub const DEFAULT_RPC_ENDPOINT: &str = "http://127.0.0.1:8545";
+/// Default Anvil chain ID
+pub const DEFAULT_CHAIN_ID: u64 = 31337;
+/// Default private key derived from starting Anvil as follows:
+/// anvil -b 5 --config-out $BUILD_DIR/anvil.json
+/// PRE_PRIVATE=$(jq -r '.private_keys[0]' $BUILD_DIR/anvil.json)
+pub const DEFAULT_PRIVATE_KEY: &str = "ac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct EthereumClientConfig {
+    #[serde(default)]
+    pub provider: EthereumProviderConfig,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub wallet: Option<EthereumWalletConfig>,
+    #[serde(default)]
+    pub contracts: StarknetContracts,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum EthereumProviderConfig {
+    Http(HttpProviderConfig),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum EthereumWalletConfig {
+    Local(LocalWalletConfig),
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct StarknetContracts {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub core_contract: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub verifier_contract: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub memory_pages_contract: Option<String>,
+}
```

Comment:
Why is it interesting to have Option there, rather than failing? 
Are there use cases where we can still have meaningful execution with one of the three missing?